waspc/src/Wasp/Generator/ServerGenerator/Common.hs

```diff
@@ -132,3 +133,8 @@ defaultServerPort = 3001
 
 defaultDevServerUrl :: String
 defaultDevServerUrl = "http://localhost:" ++ show defaultServerPort
+
+-- TODO: update this to use Wasp.SemanticVersion when we'll have support
```

Comment:
Aha! Well interesting that I haven't implemented `~` in our semver. I haven't left any TODOs there, which means I thought I implemented everything I should. Truth is, `~`, `^`, `1.2.3 - 3.4.5`, `1.2.x`, ... -> those are "sugars" (https://github.com/npm/node-semver?tab=readme-ov-file#advanced-range-syntax) on top of basic primitives. So one could say our implementation of SemanticVersion is complete but it doesn't have this "sugar" layer, these advanced ways to define ranges, so we could add that on top in the future. But then I see that I did implement `^` :D. I wonder why I implemented just that one and not the rest. Ok. Maybe you can put a TODO in SemanticVersion.hs to implement other advanced ranges (https://github.com/npm/node-semver?tab=readme-ov-file#advanced-range-syntax) besides `^`?

Reply:
I've left a comment in the SemanticVersion module 👍